src/database/arangodb/aql-generator.ts

```diff
@@ -1399,153 +1418,842 @@ register(FollowEdgeQueryNode, (node, context) => {
+
+    // the outerMapFrag will produce a list, so a simple RETURN mapFrag would result in nested lists
+    // -> we iterate over the items again to flatten the lists
+    // Note: If the relation traversal only consists of 1:1, we could theoretically use something like this:
+    //     LET root = FIRST(FOR obj IN OUTBOUND source edge_collection RETURN obj)
+    //     RETURN root.children
+    // however, that would prevent the reduce-extraction-to-projection optimization
+    // (because we would access the whole root object)
+    // -> it's better to produce the same structure as we do for 1:n relation traversals:
+    //     FOR root IN OUTBOUND source edge_collection
+    //     FOR item IN root.children
+    //     RETURN item
+    // (could also use FLATTEN(), but since we're already using nested FORs, this is probably cleaner)
+    const itemVar = aql.variable(`item`);
+    return aqlExt.parenthesizeList(
+        aql`${forStatementsFrag}`,
+        aql`FOR ${itemVar} IN ${outerMapFrag}`,
+        generateLimitClause(limitArgs) ?? aql``,
+        aql`RETURN ${itemVar}`,
+    );
+}
+
+/**
+ * Produces:
+ *
+ *     FOR node1 IN OUTBOUND source_hop1
+ *     FOR node2 in OUTBOUND hop1_hop2
+ *     FOR root IN OUTBOUND hop2_target
+ *     FOR item IN root.fieldSegment1[*].fieldSegment2[**][*
+ *         FILTER filterExpr(CURRENT)
+ *         RETURN {
+ *             value: innerNode(CURRENT),
+ *             sortValues: [
+ *                 CURRENT.sortField1,
+ *                 CURRENT.sortField2
+ *             ]
+ *         }
+ *     ]
+ *     SORT item.sortValues[0] ASC, item.sortValues[1] DESC
+ *     LIMIT skip, maxCount
+ *     RETURN item.value
+ *
+ * or, if preserveNullValues is true and hop2_target is a to-1 relation:
+ *
+ *     FOR node1 IN OUTBOUND source_hop1
+ *     FOR node2 in OUTBOUND hop1_hop2
+ *     LET root = FIRST(FOR node3 IN OUTBOUND hop2_target RETURN node3)
+ *     FOR item IN root.fieldSegment1[*].fieldSegment2[**][*
+ *         FILTER filterExpr(CURRENT)
+ *         RETURN {
+ *             value: innerNode(CURRENT),
+ *             sortValues: [
+ *                 CURRENT.sortField1,
+ *                 CURRENT.sortField2
+ *             ]
+ *         }
+ *     ]
+ *     SORT item.sortValues[0] ASC, item.sortValues[1] DESC
+ *     LIMIT skip, maxCount
+ *     RETURN item.value
+ */
+function processTraversalWithRelationAndListFieldSegmentsWithSort(
+    node: TraversalQueryNode,
+    context: QueryContext,
+) {
+    if (!node.relationSegments.length || !node.fieldSegments) {
+        throw new Error(`Expected both relation and field segments`);
+    }
+    if (!node.fieldSegments.some((s) => s.isListSegment)) {
+        throw new Error(`Expected at least one list field segment`);
+    }
+    if (!node.resultIsList) {
+        throw new Error(`Cannot have sort on non-list traversal`);
+    }
+
+    // this will hold the node of the innermost relation traversal
+    const rootVar = aql.variable('root');
```

Comment:
It might make sense to also check for `node.orderBy.clauses.length > 0` here and throw otherwise. You implicitly handle the case below because nothing goes wrong in case there are no order clauses but for the sake of completeness you could add the same check here.